doc/source/whatsnew/v0.25.0.rst

```diff
@@ -99,6 +99,7 @@ Other Enhancements
 - Error message for missing required imports now includes the original import error's text (:issue:`23868`)
 - :class:`DatetimeIndex` and :class:`TimedeltaIndex` now have a ``mean`` method (:issue:`24757`)
 - :meth:`DataFrame.describe` now formats integer percentiles without decimal point (:issue:`26660`)
+- :func:`pandas.merge` now raises ``ValueError`` when either of ``left_on`` or ``right_on`` is not provided (:issue:`26855`)
```

Comment:
can you elaborate slightly; this *only* matters if ``on`` itself is not provided. Do we have a check if ``on`` is provide and *either* of ``left_on`` or ``right_on`` is not ``None``?